src/passes/bytesConverter.ts

```diff
@@ -47,6 +53,59 @@ export class BytesConverter extends ASTMapper {
     this.commonVisit(node, ast);
   }
 
+  visitIndexAccess(node: IndexAccess, ast: AST): void {
+    if (
+      !(
+        node.vIndexExpression &&
+        generalizeType(getNodeType(node.vBaseExpression, ast.compilerVersion))[0] instanceof
+          FixedBytesType &&
+        getNodeType(node, ast.compilerVersion) instanceof FixedBytesType
+      )
+    ) {
+      const typeNode = replaceFixedBytesType(getNodeType(node, ast.compilerVersion));
```

Comment:
I think this should just be a call to visitExpression rather than copying the implementation to keep things centralised in case they need changing in the future